packages/ember-views/lib/system/event_dispatcher.js

```diff
@@ -218,43 +218,22 @@ export default EmberObject.extend({
     });
 
     rootElement.on(`${event}.ember`, '[data-ember-action]', evt => {
-      let actionId = jQuery(evt.currentTarget).attr('data-ember-action');
-      let actions = ActionManager.registeredActions[actionId];
+      let attributes = evt.currentTarget.attributes;
+      let attributeCount = attributes.length;
 
-      // In Glimmer2 this attribute is set to an empty string and an additional
-      // attribute it set for each action on a given element. In this case, the
-      // attributes need to be read so that a proper set of action handlers can
-      // be coalesced.
-      if (actionId === '') {
-        let attributes = evt.currentTarget.attributes;
-        let attributeCount = attributes.length;
+      for (let i = 0; i < attributeCount; i++) {
+        let attr = attributes.item(i);
+        let attrName = attr.name;
 
-        actions = [];
+        if (attrName.indexOf('data-ember-action-') === 0) {
+          //TODO: GJ: why does `registeredActions` contains arrays? are they always of length 1?
```

Comment:
In the pre-glimmer2 implementation it was an array because we only tagged the element with `data-ember-action="someid"` and that element could have multiple actions:

```hbs
<button {{action 'foo' on="click"}} {{action 'bar' on="doubleClick"}}></button>
```

In the glimmer 2 implementation, we have maintained it as an array but it can only ever have one entry.  

Some references:

* https://github.com/emberjs/ember.js/blob/c998c34de5eec1e5df3453b98c4c056e64b0dff1/packages/ember-glimmer/lib/modifiers/action.js#L205
* https://github.com/emberjs/ember.js/blob/c998c34de5eec1e5df3453b98c4c056e64b0dff1/packages/ember-glimmer/lib/modifiers/action.js#L43-L47

I think that `registerAction` can be simplified to something like:

```js
  registerAction(actionState) {
    let { actionId } = actionState;

    ActionManager.registeredActions[actionId] = actionState;

    return actionId;
  },
```